### PR TITLE
24423 Only redirect when route is Dashboard

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#24423

*Description of changes:*
- app version = 7.4.7
- don't redirect to new dashboard for local filings

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -450,9 +450,9 @@ export default class App extends Mixins(
 
     // now that LaunchDarkly has been updated (ie, in case of user targeting),
     // check whether to use this Entity Dashboard or the new Business Dashboard
-    if (GetFeatureFlag('use-business-dashboard')) {
+    if (GetFeatureFlag('use-business-dashboard') && (this.$route.name === Routes.DASHBOARD)) {
```
Comment (severinbeauvais) [inline src/App.vue:453]: This was behaving incorrectly when the (new) Business Dashboard UI redirected to this (old) Entity Dashboard to perform a local filing (eg, COD).

File: src/App.vue [hunk 2]
```diff
@@ -450,9 +450,9 @@ export default class App extends Mixins(
...
       const identifier = (this.businessId || this.tempRegNumber)
-      const dashboardUrl = `${this.getBusinessDashUrl}/${identifier}${this.$route.fullPath}`
+      const dashboardUrl = `${this.getBusinessDashUrl}${identifier}${this.$route.fullPath}`
```
Comment (severinbeauvais) [inline src/App.vue:455]: The old code caused a double slash between the URL and id.

**Review by JazzarKarim — APPROVED** [review]
LGTM

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-705-qcyo3ob5.web.app

**Pull request merged**